muikku-core-plugins/src/main/java/fi/otavanopisto/muikku/plugins/coursepicker/CoursePickerRESTService.java

```diff
@@ -452,7 +453,42 @@ public Response getWorkspace(@PathParam("ID") Long workspaceEntityId) {
       }
     }
     
-    return Response.ok(createRestModel(workspaceEntity, workspace.getName(), workspace.getNameExtension(), workspace.getDescription(), educationTypeName, canSignup, isCourseMember)).build();
+    return Response.ok(createRestModel(workspaceEntity, workspace.getName(), workspace.getNameExtension(), educationTypeName)).build();
+  }
+  
+  @GET
+  @Path("/workspaces/{ID}/signupDetails")
+  @RESTPermit (handling = Handling.INLINE)
+  public Response getSignupDetails(@PathParam("ID") Long workspaceEntityId) {
+    
+    WorkspaceEntity workspaceEntity = workspaceController.findWorkspaceEntityById(workspaceEntityId);
+    if (workspaceEntity == null) {
+      return Response.status(Status.NOT_FOUND).build();
+    }
+    
+    Workspace workspace = null;
+    
+    schoolDataBridgeSessionController.startSystemSession();
+    try {
+      workspace = workspaceController.findWorkspace(workspaceEntity);
+    } finally {
+      schoolDataBridgeSessionController.endSystemSession();
+    }
+
+    if (workspace == null) {
+      return Response.status(Status.NOT_FOUND).build();
+    }
+    
+    String description = workspace.getDescription();
+    boolean canSignup = getCanSignup(workspaceEntity);
+    
+    if (Boolean.TRUE.equals(canSignup)) {
+      canSignup = getIsAlreadyEvaluated(workspaceEntity);
```

Comment:
canSignup = !getIsAlreadyEvaluated(workspaceEntity);

(because of the review comment in the getIsAlreadyEvaluated method)